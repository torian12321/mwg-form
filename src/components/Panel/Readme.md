#### Basic
```
<Panel>Lorem Ipsum</Panel>
```

#### Themed
```
<Panel theme='sec'>Doler set</Panel>
```
