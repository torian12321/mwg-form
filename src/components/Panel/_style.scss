@import '../../styles/conf/index.scss';

// Vars
$panel-color-main: $themeColor_1 !default;
$panel-color-sec : $themeColor_2 !default;
$panel-font-size : $font-size !default;

// Mixins
@mixin panel_palette(
  $color-bg  : $panel-color-main,
  $color-text: smartColor($color-bg)
) {
  background: fade-out($color-bg, .15);
	color     : $color-text;
}

// Block
.panel {
  position  : relative;
  width     : 100%;
  min-height: 100px;
  padding   : 32px;
  background: $panel-color-main;
  box-sizing: content-box;
  font-size : $panel-font-size;
  box-sizing: border-box;
  @include shadow(0 0 8px 0 rgba(0, 0, 0, .4));
  @include smart-text-color($panel-color-main);

  // Modifiers
  &,
  &--main { @include panel_palette($panel-color-main, $panel-color-sec); }
  &--sec  { @include panel_palette($panel-color-sec , $panel-color-main); }
  &--white{ @include panel_palette($white); }


  @include bp-md {
    padding: 48px;
  }

  h1, h2, h3, h4, h5, h6 {
    margin-bottom: .6em;
    text-align   : center;
    font-weight  : bold;
  }
}