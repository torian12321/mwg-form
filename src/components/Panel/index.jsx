import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import './_style.scss';

class Panel extends React.Component {
  render() {
    return(
      <div className={classNames(
        'panel',
        this.props.className,
        this.props.theme && `panel--${this.props.theme}`,
      )}>
        {this.props.children}
      </div>
    )
  }
}

Panel.propTypes = {
  children : PropTypes.node.isRequired,
  className: PropTypes.string,
  theme    : PropTypes.oneOf(['main', 'sec', 'white'])
};

module.exports = Panel;
