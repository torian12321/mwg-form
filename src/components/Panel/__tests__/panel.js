import React    from 'react';
import renderer from 'react-test-renderer';
import Panel    from '../index';

describe('component', () => {
  describe('<Panel />', () => {

    it('should render correctly', () => {
      expect(renderer.create(
        <Panel>Lorem Ipsum</Panel>
      )).toMatchSnapshot();
    });

    it('Should render danger a title on header', () => {
      expect(renderer.create(
        <Panel title='danger'>Lorem Ipsum</Panel>
      )).toMatchSnapshot();
    });

  });
});
