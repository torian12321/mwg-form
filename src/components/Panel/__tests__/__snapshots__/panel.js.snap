// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Panel /> Should render danger a title on header 1`] = `
<div
  className="panel"
>
  <div
    className="panel__title"
  >
    <span>
      danger
    </span>
  </div>
  <div
    className="panel__body"
  >
    Lorem Ipsum
  </div>
</div>
`;

exports[`component <Panel /> should render correctly 1`] = `
<div
  className="panel"
>
  <div
    className="panel__body"
  >
    Lorem Ipsum
  </div>
</div>
`;
